src/Foundation/NSUrlSessionHandler.cs

```diff
@@ -542,9 +542,20 @@ InflightData GetInflightData (NSUrlSessionTask task)
 
 				lock (sessionHandler.inflightRequestsLock)
 					if (sessionHandler.inflightRequests.TryGetValue (task, out inflight)) {
-						// ensure that we did not cancel the request, if we did, do cancel the task
-						if (inflight.CancellationToken.IsCancellationRequested)
+						// ensure that we did not cancel the request, if we did, do cancel the task, if we 
+						// cancel the task it means that we are not interested in any of the delegate methods:
+						// 
+						// DidReceiveResponse     We might have received a response, but either the user cancelled or a 
+						//                        timeout did, if that is the case, we do not care about the respose.
+						// DidReceiveData         Of buffer has a partial response ergo garbage and there is not real 
```

Comment:
ergo ? we're now using latin for comments :)

Reply:
We are that posh :)